lib/polyfill/indexed_db.js

```diff
@@ -37,6 +37,17 @@ shaka.polyfill.IndexedDB = class {
       delete window.indexedDB;
       goog.asserts.assert(
           !window.indexedDB, 'Failed to override window.indexedDB');
+    } else {
+      try {
+        window.indexedDB;
+      } catch (e) {
+        if (!e instanceof SecurityError) {
```

Comment:
Actually, this seems backward.  You only remove indexedDB if it's **not** a SecurityError?

And what if it's some other error type?  It seems that if accessing the object throws, we shouldn't leave the object in place, regardless of the error type.  We would have the same problem in probeSupport().

Reply:
Sorry, my bad, fixed that typo.

The intention was to do it only if it's a SecurityError to limit the impact of this change to the described scenario.

But it's probably a good idea to catch any error and remove the object.

I've place this logic here to replicate the Chromecast case, although the exception was thrown during `probeSupport()`. Detecting the problem here was enough in the use cases I tested but I can also catch the exception in lib/offline/indexeddb/storage_mechanism.js:340 if neccesary.

Reply:
Thanks, I appreciate it.  I think this is the right place to fix it, as opposed to the offline/ folder.  The general approach we take it to polyfill broken or incompatible things, and assume that everything works and is standards-compliant in the main code.

Since this happens when the object is merely accessed, I still think I might prefer if we caught all errors.  Regardless of the error type, throwing in the getter for indexedDB is pretty severe, and it would be best if we could survive this and simply disable IDB, regardless of what type of error it is.

Reply:
Great, I've just updated the PR so any error is caught, regardless of its type.